doc/manifest/schema/1.0.0/installer.md

```diff
@@ -0,0 +1,444 @@
+## Manifest Schema v1.0.0 "installer" File
+
+All Windows Package Manager manifests in the Microsoft community repository are submitted using YAML syntax. A JSON schema is provided to aid authoring these files in editors, and in the other tooling related to the Windows Package Manager. This document provides detailed information regarding the usage of the YAML keys in the [installer](https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/manifests/v1.0.0/manifest.installer.1.0.0.json) file for multi-file manifests. Please review the [Manifest Specification](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md) if you are not familiar with this file.
+
+
+## Fields
+### PackageIdentifier
+<details>
+ <summary>The package unique identifier</summary>
+
+ #### Required Field
+
+ This key is the unique identifier for a given package. This value is generally in the form of `Publisher.Package`. It is case sensitve, and this value must match the folder structure under the partition directory in GitHub.
+
+</details>
+
+### PackageVersion
+<details>
+ <summary>The package version</summary>
+
+ #### Required Field
+
+ This key represents the version of the package. It is related to the specific release this manifests targets. In some cases you will see a perfectly formed [semantic](https://semver.org) version number, and in other cases you might see something different. These may be date driven, or they might have other characters with some package specific meaning for example.
+
+ The Windows Package Manager client uses this version to determine if an upgrade for a package is available. In some cases, packages may be released with a marketing driven version, and that causes trouble with the `winget upgrade` command. 
+
+ The current best practice is to use the value reported in Add / Remove Programs when this version of the package is installed. In some cases, packages do not report a version resulting in an upgrade loop or other unwanted behavior.
+</details>
+
+### Channel
+<details>
+ <summary>The distribution channel</summary>
+
+ #### Optional Field
+
+ This key represents the distribution channel for a package. Examples may include "stable" or "beta".
+
+ >Note: This key is included for future use. The Windows Package Manager currently does not have any behavior associated with this key. The intent behind this key is to help disambiguate the different channels for packages lacking support for side by side installation. Some packages support having more than one package channel available on a system simultaneously. This key is intended to ensure the proper channel for a package is used during install and upgrade scenarios.
+ 
+ >Note: This key may be present in the root of the manifest as the default value for all installer nodes. This key may also be present in an individual installer node as well. If this key is in the manifest root and in an installer node, the value in the installer node will apply.
```

Comment:
I would "think" we would want to have a single "package" like "Docker.Desktop" with manifests containing channels rather than two distinct packages since they cannot be installed side by side. Two packages makes sense when we can have side by side install like "Microsoft.VisualStudioCode" and "Microsoft.VisualStudioCodeInsider". We would still need a way to determine if a user had installed a particular channel so we don't do the "wrong" upgrade.

Reply:
I mean, one manifest can have only one Channel, just like one manifest can have only one Version. They still belong to same package. A different Channel most likely mean the Version is also different.

And in implementation, we already couple the Version and Channel as one concept, we operate on {Version, Channel} pair as one package version. Changing a Version to map to multiple Channels seems a lot of code rewriting involved.

Reply:
OK, so we will need some guidance on authoring manifests when channels are present **when they are supported**. We will also need some kind of identifier to identify the "default" or "stable" channel, and we will likely need some kind of description for any other channels. When we perform search, and we receive results for a package with multiple channels, we will need a way to help users identify channels. Maybe show could display channels and descriptions. We will also need a way to ensure the upgrade scenario selects the correct channel, and we will need guidance or other flags to help with migrating between channels.